Saturation/SaturationAlgorithm.cpp

```diff
@@ -524,10 +547,20 @@ void SaturationAlgorithm::onClauseReduction(Clause* cl, Clause* replacement,
   premStack.reset();
   premStack.loadFromIterator(premises);
 
+  static ClauseStack repStack;
+  repStack.reset();
+  repStack.loadFromIterator(replacements);
+
+  Clause* replacement = repStack.size() ? repStack[0] : 0;
+
   if (env.options->showReductions()) {
     env.beginOutput();
     env.out() << "[SA] " << (forward ? "forward" : "backward") << " reduce: " << cl->toString() << endl;
-    if(replacement){ env.out() << "     replaced by " << replacement->toString() << endl; }
+    ClauseStack::Iterator rit(repStack);
+    while(rit.hasNext()){
+      Clause* replacement = rit.next();
+      if(replacement){ env.out() << "      replaced by " << replacement->toString() << endl; }
+    }
```

Comment:
Is the use case for multiple replacements higher-order logic only?

Reply:
Correct. We currently have higher-order immediate simplifcation inferences that return multiple conclusions. 

It could also be used for `InjectivityGIE` in `TermAlgebraReasoning` which has been inaccurately (in comparison to the paper) coded as a generating inference. However, this hasn't been done yet.